src/views/blog-post/table-of-contents/heading-intersection-observer-script.astro

```diff
@@ -24,7 +24,7 @@ const { headingsToDisplaySlugs } = Astro.props as {
 		const tocListContainer = tocListRef.parentElement!.parentElement!;
 
 		const linkRefs: LinkRef[] = [
-			...(document.querySelectorAll(
+			...(tocListRef.querySelectorAll(
```

Comment:
This makes sure the linkRefs is only populated with items inside the list component, rather than the whole document.